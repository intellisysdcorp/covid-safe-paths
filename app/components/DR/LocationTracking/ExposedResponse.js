import { Button, Container, Content, Text } from 'native-base';
import React from 'react';
import { ScrollView, View } from 'react-native';
import {
  heightPercentageToDP as hp,
  widthPercentageToDP as wp,
} from 'react-native-responsive-screen';

import Header from '../../../components/DR/Header';
import styles from '../../../components/DR/Header/style';
import ResultsContent from '../../../components/DR/ResultsContent';
import Colors from '../../../constants/colors';
import languages from '../../../locales/languages';

export default function ExposedResponse({ navigation }) {
  return (
    <Container>
      <Content>
        <ScrollView>
          <View style={{ flex: 1 }}>
            <Header
              title={languages.t('label.exposure_next_steps')}
              text='No ha tenido síntomas de COVID-19, pero contacto expuesto a alguien con COVID-19.'
              navigation={navigation}
              style={{ height: hp('18%') }}
            />
            <View style={styles.formContainer}>
              {
                <ResultsContent
                  title='Deberías practicar distancia social'
                  subtitle='En este momento, usted no presenta síntomas de COVID-19, pero ha estado expuesto a un caso, por lo que le recomendamos que continúe con las medidas preventivas de “lavado de manos con agua y jabón de manera frecuente, aplicar el distanciamiento social” y aplicar las orientaciones básicas para el hogar y la comunidad: '
                  subLinkText='http://digepisalud.gob.do/documentos'
                  subLink='http://digepisalud.gob.do/documentos/?drawer=Vigilancia%20Epidemiologica*Alertas%20epidemiologicas*Coronavirus*Nacional*Materiales%20IEC%20COVID-19'
                  image={require('../../../assets/images/socialDistancing.jpg')}
                  navigation={navigation}
                  nextSteps={[
                    {
                      title: 'Cuarentena en casa',
                      content:
                        'Es posible que haya estado expuesto. Debe quedarse en casa durante los próximos 14 días y ver si aparece algún síntoma. Manténgase al menos a un (1) metro de distancia de cualquier persona y evite estar en grupos.',
                    },
                    {
                      title: 'Monitorear síntomas',
                      content:
                        'Esté atento a los síntomas de COVID-19, como tos, fiebre y dificultad para respirar. Además, revise y mida su temperatura dos veces al día durante dos semanas. Si presenta algún síntoma de peligro (vómito o diarrea persistente o con sangre, dificultad marcada para respirar, somnolencia o convulsiones) acuda de forma inmediata al centro de salud más cercano o llame al *462.',
                    },
                  ]}
                  recomendations={[
                    {
                      title: 'No se necesita prueba en este momento',
                      content:
                        'A partir de ahora, sus respuestas sugieren que no necesita hacerse la prueba. Si llega a presentar algún síntoma como fiebre, tos, secreción nasal, ligera dificultad para respirar, dolor de garganta, cabeza o del cuerpo, vuelva a completar este cuestionario  o contáctese con la plataforma Aurora a través de los teléfonos 809-449-6262, 809 409 6262 y 809-352-6262.',
                    },
                    {
                      title:
                        '“Con tu ayuda, quedandote en casa y siguiendo las recomendaciones de salud, ponemos enfrentar al COVID-19”',
                      content: '',
                    },
                  ]}
                />
              }
              <Button
                style={[
                  styles.buttons,
                  {
                    width: wp('70%'),
                    height: 38,
                    backgroundColor: Colors.GREEN,
                    marginTop: 15,
                  },
                ]}
                onPress={() => {
                  navigation.navigate('HomeScreen');
                }}>
                <Text style={[styles.buttonText, { color: '#fff' }]}>
                  Cerrar
                </Text>
              </Button>
            </View>
          </View>
        </ScrollView>
      </Content>
    </Container>
  );
}
