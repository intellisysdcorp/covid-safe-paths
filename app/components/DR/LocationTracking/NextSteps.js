import React, { Component } from 'react';
import { withTranslation } from 'react-i18next';
import { StyleSheet } from 'react-native';

import { Button } from '../../../components/Button';

class NextSteps extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    const { t } = this.props;
    let buttonLabel;
    let buttonFunction;
    buttonLabel = t('label.exposure_next_steps');
    buttonFunction = () => {
      this.props.navigation.navigate('ExposedResponse');
    };
    return (
      <Button
        label={buttonLabel}
        onPress={() => buttonFunction()}
        style={styles.buttonContainer}
        secondary
      />
    );
  }
}

const styles = StyleSheet.create({
  buttonContainer: {
    marginTop: 24,
    height: 54, // fixes overlaying buttons on really small screens
  },
});

export default withTranslation()(NextSteps);
