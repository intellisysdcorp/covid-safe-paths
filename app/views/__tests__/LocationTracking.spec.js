import 'react-native';

import BackgroundGeolocation from '@mauron85/react-native-background-geolocation';
import { act, render } from '@testing-library/react-native';
import React from 'react';

import LocationTracking from '../LocationTracking';

const navigationMock = {
  addListener: jest.fn(),
};

jest.mock('../../constants/DR/baseUrls', () => ({
  MEPYD_C5I_API_URL: 'contact_tracing/api',
  MEPYD_C5I_SERVICE: 'https://webapps.mepyd.gob.do',
  GOV_DO_TOKEN: '',
}));

beforeEach(() => {
  BackgroundGeolocation.checkStatus.mockReturnValue(jest.fn());
  BackgroundGeolocation.configure.mockReturnValue(jest.fn());
});

it('renders correctly', async () => {
  const { asJSON } = render(<LocationTracking navigation={navigationMock} />);

  await act(async () => {
    expect(asJSON()).toMatchSnapshot();
  });
});
