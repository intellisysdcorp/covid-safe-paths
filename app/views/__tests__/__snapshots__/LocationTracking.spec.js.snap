// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "flex": 1,
        "height": "100%",
        "justifyContent": "flex-end",
        "resizeMode": "cover",
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../app/assets/images/launchScreenBackground.png",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": "100%",
            "width": "100%",
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#0161F2",
          "flex": 1,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "resizeMode": "contain",
          "right": 0,
          "top": "-5%",
        }
      }
    >
      <SvgXml
        height={1334}
        width={1334}
        xml="<svg width=\\"850\\" height=\\"850\\" viewBox=\\"0 0 850 850\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\">
<path d=\\"M425 465C447.091 465 465 447.091 465 425C465 402.909 447.091 385 425 385C402.909 385 385 402.909 385 425C385 447.091 402.909 465 425 465Z\\" fill=\\"white\\"/>
<path d=\\"M425 457C442.673 457 457 442.673 457 425C457 407.327 442.673 393 425 393C407.327 393 393 407.327 393 425C393 442.673 407.327 457 425 457Z\\" fill=\\"#41DCA4\\"/>
<path d=\\"M425 412C417.832 412 412 417.832 412 425C412 432.168 417.832 438 425 438C432.168 438 438 432.168 438 425C438 417.832 432.168 412 425 412ZM425 435.075C419.436 435.075 414.925 430.565 414.925 425C414.925 419.436 419.436 414.925 425 414.925C430.564 414.925 435.075 419.436 435.075 425C435.075 430.565 430.564 435.075 425 435.075Z\\" fill=\\"white\\"/>
<path d=\\"M430.844 419.727C430.22 419.215 429.298 419.305 428.786 419.93L423.15 426.797L421.24 424.334C420.745 423.695 419.826 423.58 419.188 424.074C418.55 424.57 418.433 425.488 418.928 426.127L421.962 430.039C422.235 430.391 422.653 430.599 423.098 430.605C423.104 430.606 423.111 430.606 423.118 430.606C423.556 430.606 423.97 430.409 424.249 430.071L431.047 421.785C431.559 421.161 431.469 420.239 430.844 419.727Z\\" fill=\\"white\\"/>
<circle opacity=\\"0.2\\" cx=\\"425\\" cy=\\"425\\" r=\\"85.5\\" stroke=\\"white\\"/>
<circle opacity=\\"0.2\\" cx=\\"425\\" cy=\\"425\\" r=\\"149.5\\" stroke=\\"white\\"/>
<circle opacity=\\"0.2\\" cx=\\"425\\" cy=\\"425\\" r=\\"217.5\\" stroke=\\"white\\"/>
<circle opacity=\\"0.2\\" cx=\\"425\\" cy=\\"425\\" r=\\"283.5\\" stroke=\\"white\\"/>
<circle opacity=\\"0.2\\" cx=\\"425\\" cy=\\"425\\" r=\\"352.5\\" stroke=\\"white\\"/>
<circle opacity=\\"0.2\\" cx=\\"425\\" cy=\\"425\\" r=\\"424.5\\" stroke=\\"white\\"/>
</svg>"
      />
    </View>
    <View
      style={
        Object {
          "height": "100%",
          "left": 0,
          "paddingBottom": 12,
          "paddingHorizontal": "12%",
          "position": "absolute",
          "right": 0,
          "top": "5%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "flex-end",
            "paddingBottom": 80,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFF",
              "fontFamily": "IBMPlexSans-Light",
              "fontSize": 16,
              "fontWeight": "normal",
              "lineHeight": 24.5,
              "marginBottom": 24,
              "textAlign": "center",
              "top": "-15%",
              "writingDirection": "ltr",
            }
          }
          use="body1"
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "flex-start",
            "paddingTop": 80,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFF",
              "fontFamily": "IBMPlexSans-Medium",
              "fontSize": 26,
              "fontWeight": "normal",
              "lineHeight": 34,
              "marginBottom": 24,
              "textAlign": "center",
              "writingDirection": "ltr",
            }
          }
          use="body1"
        >
          No known contact
        </Text>
        <Text
          style={
            Object {
              "color": "#FFF",
              "fontFamily": "IBMPlexSans",
              "fontSize": 18,
              "fontWeight": "normal",
              "lineHeight": 24.5,
              "marginBottom": 24,
              "textAlign": "center",
              "writingDirection": "ltr",
            }
          }
          use="body1"
        >
          Based on available data you havenâ€™t been near anyone reported positive for COVID-19.
        </Text>
      </View>
    </View>
    <View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        style={
          Object {
            "backgroundColor": "#0161F2",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignContent": "flex-end",
              "flexDirection": "column",
              "justifyContent": "space-between",
              "padding": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#E5E4E6",
                "fontFamily": "IBMPlexSans-Bold",
                "fontSize": 18,
                "fontWeight": "normal",
                "lineHeight": 24,
                "textAlign": "left",
                "writingDirection": "ltr",
              }
            }
            use="body1"
          >
            More COVID-19 information
          </Text>
          <Text
            style={
              Object {
                "color": "#E5E4E6",
                "fontFamily": "IBMPlexSans",
                "fontSize": 18,
                "fontWeight": "normal",
                "lineHeight": 24,
                "textAlign": "left",
                "writingDirection": "ltr",
              }
            }
            use="body1"
          >
            from the Mayo Clinic
          </Text>
        </View>
        <View
          style={
            Object {
              "alignSelf": "center",
              "paddingLeft": 20,
              "paddingRight": 20,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../app/assets/images/foreArrow.png",
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      style={
        Object {
          "alignSelf": "flex-end",
          "marginRight": "7%",
          "marginTop": "14%",
          "opacity": 1,
          "position": "absolute",
          "top": 0,
        }
      }
    >
      <Image
        resizeMode="contain"
      />
      <SvgXml
        color="white"
        height={30}
        width={30}
        xml="
<svg width=\\"24\\" height=\\"22\\" viewBox=\\"0 0 24 22\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\">
  <path d=\\"M23.625 17H9V15.125C9 14.5156 8.48438 14 7.875 14H5.625C4.96875 14 4.5 14.5156 4.5 15.125V17H0.375C0.140625 17 0 17.1875 0 17.375V18.125C0 18.3594 0.140625 18.5 0.375 18.5H4.5V20.375C4.5 21.0312 4.96875 21.5 5.625   21.5H7.875C8.48438 21.5 9 21.0312 9 20.375V18.5H23.625C23.8125 18.5 24 18.3594 24 18.125V17.375C24 17.1875 23.8125 17 23.625 17ZM7.5 20H6V15.5H7.5V20ZM23.625 3.5H12V1.625C12 1.01562 11.4844 0.5 10.875 0.5H8.625C7.96875 0.5 7.5 1.01562 7.5 1.625V3.5H0.375C0.140625 3.5 0 3.6875 0 3.875V4.625C0 4.85938 0.140625 5 0.375 5H7.5V6.875C7.5 7.53125 7.96875 8 8.625 8H10.875C11.4844 8 12 7.53125 12 6.875V5H23.625C23.8125 5 24 4.85938 24 4.625V3.875C24 3.6875 23.8125 3.5 23.625 3.5ZM10.5 6.5H9V2H10.5V6.5ZM23.625 10.25H19.5V8.375C19.5 7.76562 18.9844 7.25 18.375 7.25H16.125C15.4688 7.25 15 7.76562 15 8.375V10.25H0.375C0.140625 10.25 0 10.4375 0 10.625V11.375C0 11.6094 0.140625 11.75 0.375 11.75H15V13.625C15 14.2812 15.4688 14.75 16.125 14.75H18.375C18.9844 14.75 19.5 14.2812 19.5 13.625V11.75H23.625C23.8125 11.75 24 11.6094 24 11.375V10.625C24 10.4375 23.8125 10.25 23.625 10.25ZM18 13.25H16.5V8.75H18V13.25Z\\" fill=\\"currentColor\\" />
</svg>
"
      />
    </View>
  </View>
</View>
`;
